solver-next/types.go

```diff
@@ -31,6 +30,14 @@ type Edge struct {
 	Vertex Vertex
 }
 
+// Metadata has optional metadata for the vertex that is not contained in digest
+type Metadata struct {
+	IgnoreCache bool
+	CacheSource CacheManager
```

Comment:
I'd expect Metadata to only contain serializable data

Reply:
How would you solve this problem then as we can't put a registry reference to here and load it for every vertex separately? The `pb.Metadata` is completely serializable but `Vertex/Edge` itself are only linked objects in memory(without any serializable data to even verify the digest for example) so I don't see why this restriction is needed.

Btw, I forgot to add this field to `llb.Load`. That function should take options, one of them is `WithCache` and this is where this would be filled. If there is a use-case (for example for subbuilds) we could add a cache source reference to `pb.Metadata` and convert these on `llb.Load` as well.

If you are ok with this, I can make the `llb.Load` change in this PR. I can't make the other one yet as there is no (new) way to convert a reference to new cache interface yet.

Reply:
How about calling it `VertexOptions`, as `CacheSource` seems kind of procedure rather than "data".

I'm also planning to replace the `IgnoreCache bool` metadata with more flexible `func() bool` procedure.

Reply:
What is the use-case you have to the `IgnoreCache` callback? To implement it I think I need to evaluate it in load time instead of while it is executing to handle the case where to jobs build the same vertex with different ignore-cache value. So wanted to make sure this still works for you?

Reply:
e.g. "expire cache if it was created before yyyyMMdd"

Reply:
ah, that would be ok I evaluated on load time